app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ReadFileTask.java

```diff
@@ -126,6 +127,9 @@ protected ReturnedValues doInBackground(Void... params) {
     } catch (IOException e) {
       e.printStackTrace();
       return new ReturnedValues(EXCEPTION_IO);
+    } catch (OutOfMemoryError e) {
+      e.printStackTrace();
+      return new ReturnedValues(EXCEPTION_OOM);
```

Comment:
How will it create a new object if it has no more memory?

Reply:
> How will it create a new object if it has no more memory?

How else? Any suggestions?

Reply:
* You can check if you have enough memory _before_ trying to reserve it.
* You can use a window based system to access the file intelligently (I know that search is an issue, but you can simply disable search for now if you find that using another window to search is too complicated).
* You can use a `malloc` like system that allows for a "null" response if you can't allocate the memory, and that basically checks if you have the memory before trying to allocate.

Reply:
* You can load memory in `chunks` and stop if onLowMemory or onTrimMemory are called.